BoneCentral/Brain/Utilities/compassTools.js

```diff
@@ -0,0 +1,51 @@
+var vector = require('./vector.js');
+
+var getTrueCompass = function(compv, accelv) {
+    var av      = unitScale(accelv);
+    var right   = unitScale(vector.cross(av, compv));
+    var res     = unitScale(vector.cross(right, av));
+    return res;
+};
+
+var cdeg = function(compv, va) {
```

Comment:
Yes; I was problem solving concurrently with the writing of this code and so many of the variables I was writing, deleting, and rewriting over and over. This lead me to abbreviating the names to speed up development. Unfortunately I never took the time to expand them.

`cdeg` stands for compass degrees
`av` - acceleration vector
`right` - right vector (as in "look", "right", "up" - common in game-dev programming and 3D graphics)
`res` - result
`va` - also acceleration vector (named this way to avoid being confused with the next abbreviations)
`az` - z-axis vector
`ax` - x-axis vector
`axis` - the axis to rotate around (aka rotational axis)
`angy` - the angle between the x-axis and the rotational axis
`angx` - the angle between the acceleration-vector and the z-axis (I can't remember exactly why I used x and y. Probably something to do with the orientation of rotations.)
`c` - can't remember what this vector was (something about an intermediate step toward the correct compass vector)
`cv` - the (corrected) compass vector